docs/rfc/0005_scan_filter.md

```diff
@@ -0,0 +1,113 @@
+
+## Examples / User Stories
+
+[examples]: #examples
+
+- As a user, I want to scan only `linux/amd64` based images on my registry.
+- As a user, I want to scan `linux/amd64` and `linux/arm` based images on my registry.
+- As a user, I want to scan only the most recent tags of images on my registry.
+- As a user, I want to scan only the most recent tags, `linux/amd64` based images on my registry.
+- As a user, I want to scan only for non release candidate tags.
+- As a user, I want to scan for images deployed in a production environment with tags that end in `-prod`.
+
+# Detailed design
+
+[design]: #detailed-design
+
+## Platforms
+
+The platform filters will be used to avoid the discovery of image platforms on the registry. This will save us several calls to the registry since to do so, we have to loop over each image found in there and retrieve its details to inspect the platform.
+
+We will provide the user with a new field on the `Registry` CRD called `platforms`. The field will contain the list of platforms allowed.
+
+The `Platoform` type is a structure of 3 elements to describe it with the following elements:
+
+* `os`: specifies the operating system, for example `linux` or `windows`.
+
+* `arch`: specifies the CPU architecture.
+
+* `variant`: is an optional field specifying a variant of the CPU.
+
+Here's an example of `Registry` configured to scan only for `linux/amd64` and `linux/arm/v7` platforms:
+
+```yaml
+apiVersion: sbomscanner.kubewarden.io/v1alpha1
+kind: Registry
+metadata:
+  name: my-first-registry
+  namespace: default
+spec:
+  uri: dev-registry.default.svc.cluster.local:5000
+  platforms:
+    - arch: "amd64"
+      os: "linux"
+    - arch: "arm"
+      os: "linux"
+      variant: "v7"
+```
+
+A `platforms` field not set means that no filters are going to be applied to the scan. This will result in a complete scan with image catalogization.
+
+## Tags
+
+The tags filter will be used to avoid scanning unwanted tags for the given image. This will help the user focus only on the tags they really want to scan. We will leverage the CEL expression language to filter for image tags, enabling support for the semver library.
+
+We will provide the user with a new field on the `Registry` CRD called `matchConditions`, under the `repositories` field. For this feature, we are going to introduce a breaking change, since `repositories` is currently a `string`, and we are going to replace it with a new struct to host the `name` of the repository and the `matchConditions` struct. The `matchConditions` field will contain an arbitrary list of CEL expressions to define with high precision the list of tags to scan.
```

Comment:
We're missing:
- Link to CEL Kubernetes documentation
- Which operations we support (string, regex, and semver from Kubernetes)

Add these links:
- CEL introduction: https://kubernetes.io/docs/reference/using-api/cel/
- CEL language repository: https://github.com/google/cel-go
- CEL spec repository: https://github.com/google/cel-spec
- String functions: https://github.com/google/cel-spec/blob/master/doc/langdef.md#string-functions
- Semver operations: https://kubernetes.io/docs/reference/using-api/cel/#kubernetes-semver-library